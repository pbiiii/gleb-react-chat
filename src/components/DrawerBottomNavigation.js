import React from 'react';
import PropTypes from 'prop-types';
import RestoreIcon from '@material-ui/icons/Restore';
import ExploreIcon from '@material-ui/icons/Explore';
import { BottomNavigation, BottomNavigationAction } from '@material-ui/core';

const DrawerBottomNavigation = ({ activeTab, onTabChange }) => (
    <BottomNavigation showLabels value={activeTab} onChange={onTabChange}>
        <BottomNavigationAction label="My Chats" icon={<RestoreIcon />} value="my" />
        <BottomNavigationAction label="Explore" icon={<ExploreIcon />} value="all" />
    </BottomNavigation>
);

DrawerBottomNavigation.propTypes = {
    activeTab: PropTypes.string.isRequired,
    onTabChange: PropTypes.func.isRequired,
};

export default DrawerBottomNavigation;
