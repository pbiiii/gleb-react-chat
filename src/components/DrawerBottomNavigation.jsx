import React from 'react';
import RestoreIcon from '@material-ui/icons/Restore';
import ExploreIcon from '@material-ui/icons/Explore';
import { BottomNavigation, BottomNavigationAction } from '@material-ui/core';

const DrawerBottomNavigation = ({ activeTab, onTabChange }) => (
    <BottomNavigation
        showLabels
        value={activeTab}
        onChange={onTabChange}
    >
        <BottomNavigationAction
            label="My Chats"
            icon={<RestoreIcon />}
            value="my"
        />
        <BottomNavigationAction
            label="Explore"
            icon={<ExploreIcon />}
            value="all"
        />
    </BottomNavigation>
);

export default DrawerBottomNavigation;
