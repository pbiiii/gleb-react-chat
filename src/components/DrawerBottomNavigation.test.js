/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import DrawerBottomNavigation from './DrawerBottomNavigation';

const mockProps = {
    activeTab: 'my',
    onTabChange: () => {},
};

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<DrawerBottomNavigation {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('render correctly', () => {
    const tree = renderer
        .create(<DrawerBottomNavigation {...mockProps} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
