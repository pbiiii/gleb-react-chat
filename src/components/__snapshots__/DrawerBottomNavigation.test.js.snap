// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<div
  className="MuiBottomNavigation-root-26"
>
  <button
    className="MuiButtonBase-root-32 MuiBottomNavigationAction-root-27 MuiBottomNavigationAction-selected-28"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiBottomNavigationAction-wrapper-30"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-35"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M13 3c-4.97 0-9 4.03-9 9H1l3.89 3.89.07.14L9 12H6c0-3.87 3.13-7 7-7s7 3.13 7 7-3.13 7-7 7c-1.93 0-3.68-.79-4.94-2.06l-1.42 1.42C8.27 19.99 10.51 21 13 21c4.97 0 9-4.03 9-9s-4.03-9-9-9zm-1 5v5l4.28 2.54.72-1.21-3.5-2.08V8H12z"
        />
      </svg>
      <span
        className="MuiBottomNavigationAction-label-31 MuiBottomNavigationAction-selected-28"
      >
        My Chats
      </span>
    </span>
    <span
      className="MuiTouchRipple-root-44"
    />
  </button>
  <button
    className="MuiButtonBase-root-32 MuiBottomNavigationAction-root-27"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiBottomNavigationAction-wrapper-30"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-35"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 10.9c-.61 0-1.1.49-1.1 1.1s.49 1.1 1.1 1.1c.61 0 1.1-.49 1.1-1.1s-.49-1.1-1.1-1.1zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm2.19 12.19L6 18l3.81-8.19L18 6l-3.81 8.19z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
      <span
        className="MuiBottomNavigationAction-label-31"
      >
        Explore
      </span>
    </span>
    <span
      className="MuiTouchRipple-root-44"
    />
  </button>
</div>
`;
